### PR TITLE
Refactor net base for pipeline

### PR DESCRIPTION
**Pull request by Sammers21**
Implementation of #12 issue

### CODE REVIEW

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 1]
```diff
@@ -471,34 +472,18 @@ void publishRelease(int publishMessageId) {
...
-    if (options.getLogActivity()) {
-      pipeline.addLast("logging", new LoggingHandler());
-    }
+  private void initChannel(ChannelPipeline pipeline) {
 
     // add into pipeline netty's (en/de)coder
-    pipeline.addLast("mqttEncoder", MqttEncoder.INSTANCE);
-    pipeline.addLast("mqttDecoder", new MqttDecoder());
+    pipeline.addBefore("handler", "mqttEncoder", MqttEncoder.INSTANCE);
+    pipeline.addBefore("handler", "mqttDecoder", new MqttDecoder());
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: why did you change the handlers name (i.e. "mqttEncoder" -> "handler"). They are all called "handler" now. Is there a specific reason for that ? And why did you change from addLast to addBefore ?

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 2]
```diff
@@ -471,34 +472,18 @@ void publishRelease(int publishMessageId) {
...
 
     if (this.options.isAutoKeepAlive() &&
       this.options.getKeepAliveTimeSeconds() != 0) {
 
-      pipeline.addLast("idle",
+      pipeline.addBefore("handler", "idle",
         new IdleStateHandler(0, this.options.getKeepAliveTimeSeconds(), 0));
-      pipeline.addLast("keepAliveHandler", new ChannelDuplexHandler() {
+      pipeline.addBefore("handler", "keepAliveHandler", new ChannelDuplexHandler() {
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: as above, regarding handlers name and addLast vs addBefore

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java (hunk 1 above)
Comment (Sammers21) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: Yes, the reason is that according to the [NetSocketInternal source code](https://github.com/eclipse/vert.x/blob/master/src/main/java/io/vertx/core/impl/NetSocketInternal.java) the recommended way to add handlers into pipeline is something like this 
```java
ChannelPipeline pipeline = so.channelHandlerContext().pipeline();
pipeline.addBefore("handler", "myhandler", new MyHandler());
```

Also, I have tested the old approach and all tests failed with TimeoutException

Comment (Sammers21) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: One more argument for addBefore: in vertx-mqtt-server sources a addBefore also used when adding handlers into pipeline. See: https://github.com/vert-x3/vertx-mqtt-server/blob/master/src/main/java/io/vertx/mqtt/impl/MqttServerImpl.java#L79

Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: You are right.

**Comment by Sammers21** [comment]
Sure, commits are squashed
